api/v1alpha1/envoypatchpolicy_types.go

```diff
@@ -0,0 +1,124 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gwapiv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+const (
+	// KindEnvoyPatchPolicy is the name of the EnvoyPatchPolicy kind.
+	KindEnvoyPatchPolicy = "EnvoyPatchPolicy"
+)
+
+// +kubebuilder:object:root=true
+
+// EnvoyPatchPolicy allows the user to modify the generated Envoy xDS
+// resources by Envoy Gateway using this patch API
+type EnvoyPatchPolicy struct {
```

Comment:
Why name it a "Policy" instead of just a "Patch"?

Reply:
previous reviewers for the RateLimit feature had requested RateLimit to suffix with Filter so we could differentiate Filter and Policy